config/initializers/dfe_sign_in.rb

```diff
@@ -0,0 +1,5 @@
+DfeSignIn.configure do |config|
+  config.client_id = ENV.fetch("DFE_SIGN_IN_API_CLIENT_ID")
+  config.secret = ENV.fetch("DFE_SIGN_IN_API_SECRET")
+  config.base_url = ENV.fetch("DFE_SIGN_IN_API_ENDPOINT")
```

Comment:
Do you think we should be using `fetch` here? At the moment it's possible to run development without the need for DfE environment vars. This change would require all devs to at least put empty variables in their `.env`. Or perhaps we could reintroduce the shared `.env.development` file? (It will probably be reintroduced as part of #179 anyway)

Reply:
A shared `.env.development` that covered everything needed to run the app locally will be hard because we need secrets in it, for example `DFE_SIGN_IN_API_SECRET` and we wouldn't want to include those in the repo.

I think it's reasonable to expect developers to manage/maintain the environment variables needed to run the application. Using `fetch` here makes it super-explicit what is missing as well, making it easier to know what needs adding.

Reply:
Yeah I was thinking a shared `.env.development` with empty values so you could at least run the app. I think we may need to amend the README as it makes these variables out to be optional

Reply:
I'm not convinced a `.env.development` file in the repo with incomplete variables is a good idea.  Firstly it'll be impossible to update it without Git picking up the changes. What does a developer do in that situation? Ignore the changes when they create their commits? Add it to their local ignore file, but then remember to remove it when they need to make legit changes to it?

Also, yes the application will run, but it will be broken – when a developer attempts to authenticate with DfE Sign-in they will get a confusing error:

```DfeSignIn::ExternalServerError  – 403: {"success":false,"message":"invalid signature"}```

A developer would have to figure out that it's because of the empty environment variable, and then also figure out how to set the value without committing changes to `.env.development`.

I much prefer the developer seeing an early an explicit `key not found: "DFE_SIGN_IN_API_SECRET"` error on launch. OK, it means developers will need to set the environment variables after this is merged, but I think that one-time cost outweighs the cost of an incomplete developer env file in the repo.

If we do want to sign-post the environment variables beyond what is already in the `README`, one thing we could do is add an `.env.example` file for developers to copy and fill in the gaps when they first clone the repo.

Reply:
I think my contention mainly stems from this code: https://github.com/DFE-Digital/dfe-teachers-payment-service/blob/master/config/environments/development.rb#L69-L79

Previously it had been possible to run the app in development with OmniAuth in test/mock mode which I've just realised will no longer be possible after this change is in. We should probably remove that code at some point and update the docs, but happy to address that separately.

Reply:
Ah I wasn't aware of that code, good spot. Yep, we should definitely remove that as we can no longer rely on just mocking the OpenID bit, we'd also have to mock out the API call that checks for the role.